scripts/add.tokens.erc20.mjs

```diff
@@ -0,0 +1,264 @@
+		contract.name(),
+		contract.symbol(),
+		contract.decimals()
+	]).catch((err) => {
+		console.error(`Error fetching token details:\n${err}`);
+		process.exit(1);
+	});
+
+	return { name, symbol, decimals };
+};
+
+const createEnvFile = async (tokenName, tokenDetails, contractAddress, testnetContractAddress) => {
+	const newFileName = `tokens.${tokenName.toLowerCase()}.env.ts`;
+	const newFilePath = path.join(SRC_PATH, newFileName);
+
+	try {
+		await fs.access(newFilePath);
+		console.log(`File ${newFilePath} already exists.`);
+		return;
+	} catch (err) {
+		// File does not exist, proceed with creation
+	}
+
+	const mainnetToken = `${tokenDetails.symbol}_TOKEN`;
+	const testnetToken = `SEPOLIA_${tokenDetails.symbol}_TOKEN`;
+
+	const newContent = `import { ETHEREUM_NETWORK, SEPOLIA_NETWORK } from '$env/networks.env';
+import type { RequiredErc20Token } from '$eth/types/erc20';
+import ${tokenName.toLowerCase()} from '$icp-eth/assets/${tokenName.toLowerCase()}.svg';
+
+export const ${tokenDetails.symbol}_DECIMALS = ${tokenDetails.decimals};
+
+export const ${tokenDetails.symbol}_SYMBOL = '${tokenDetails.symbol}';
+
+export const ${tokenDetails.symbol}_TOKEN_ID: unique symbol = Symbol(${tokenDetails.symbol}_SYMBOL);
+
+export const ${mainnetToken}: RequiredErc20Token = {
+	id: ${tokenDetails.symbol}_TOKEN_ID,
+	network: ETHEREUM_NETWORK,
+	standard: 'erc20',
+	category: 'default',
+	name: '${tokenDetails.name}',
+	symbol: '${tokenDetails.symbol}',
+	decimals: ${tokenDetails.symbol}_DECIMALS,
+	icon: ${tokenName.toLowerCase()},
+	address: '${contractAddress}',
+	exchange: 'erc20',
+	twinTokenSymbol: 'ck${tokenDetails.symbol}'
+};
+
+export const SEPOLIA_${tokenDetails.symbol}_SYMBOL = 'Sepolia${tokenDetails.symbol}';
+
+export const SEPOLIA_${tokenDetails.symbol}_TOKEN_ID: unique symbol = Symbol(SEPOLIA_${tokenDetails.symbol}_SYMBOL);
+
+export const ${testnetToken}: RequiredErc20Token = {
+	id: SEPOLIA_${tokenDetails.symbol}_TOKEN_ID,
+	network: SEPOLIA_NETWORK,
+	standard: 'erc20',
+	category: 'default',
+	name: '${tokenDetails.name}',
+	symbol: '${tokenDetails.symbol}',
+	decimals: ${tokenDetails.symbol}_DECIMALS,
+	icon: ${tokenName.toLowerCase()},
+	address: '${testnetContractAddress}',
+	exchange: 'erc20',
+	twinTokenSymbol: 'ckSepolia${tokenDetails.symbol}'
+};
+`;
+
+	await fs.writeFile(newFilePath, newContent);
+	console.log(`Created ${newFilePath}`);
+
+	return { newFileName, mainnetToken, testnetToken };
+};
+
+const updateTokensErc20Env = async (newFileName, mainnetToken, testnetToken) => {
+	const filePath = path.join(SRC_PATH, 'tokens.erc20.env.ts');
+	let content = await fs.readFile(filePath, 'utf8');
+
+	const mainnetRegex = /const ERC20_TWIN_TOKENS_MAINNET: RequiredErc20Token\[] = \[([\s\S]*?)];/;
+	const sepoliaRegex = /const ERC20_TWIN_TOKENS_SEPOLIA: RequiredErc20Token\[] = \[([\s\S]*?)];/;
+
+	const mainnetMatch = content.match(mainnetRegex);
+	const sepoliaMatch = content.match(sepoliaRegex);
+
+	if (mainnetMatch) {
```

Comment:
this block of code and next one are similar, maybe worth extracting a local function?

Reply:
done